analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/DisablingRequestValidationBase.cs

```diff
@@ -0,0 +1,74 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2021 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using SonarAnalyzer.Common;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Rules
+{
+    public abstract class DisablingRequestValidationBase : SonarDiagnosticAnalyzer
+    {
+        internal const string DiagnosticId = "S5753";
+        private const string MessageFormat = "Make sure disabling ASP.NET Request Validation feature is safe here.";
+
+        private readonly IAnalyzerConfiguration analyzerConfiguration;
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        protected DiagnosticDescriptor Rule { get; }
+
+        protected DisablingRequestValidationBase(System.Resources.ResourceManager rspecResources, IAnalyzerConfiguration analyzerConfiguration)
+        {
+            Rule = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, rspecResources).WithNotConfigurable();
+            this.analyzerConfiguration = analyzerConfiguration;
+        }
+
+        protected override void Initialize(SonarAnalysisContext context) =>
+            context.RegisterSymbolAction(c =>
+                {
+                    analyzerConfiguration.Initialize(c.Options);
+                    if (!analyzerConfiguration.IsEnabled(DiagnosticId))
+                    {
+                        return;
+                    }
+
+                    var attributes = c.Symbol.GetAttributes();
+                    if (attributes.IsEmpty)
+                    {
+                        return;
+                    }
+                    var attributeWithFalseParameter = attributes.FirstOrDefault(a =>
+                        a.ConstructorArguments.Length == 1
+                        && a.ConstructorArguments[0].Kind == TypedConstantKind.Primitive
+                        && a.ConstructorArguments[0].Value is bool b
+                        && b == false
+                        && a.AttributeClass.Is(KnownType.System_Web_Mvc_ValidateInputAttribute));
```

Comment:
I'd check the `AttributeClass` first, to follow the semantics. We're looking for `ValidateInput attribute with one argument that has value False`.

We're already working with symbol information, so there's no performance penalty.

Reply:
it's a micro-optimization, but [IsMatch](https://github.com/SonarSource/sonar-dotnet/blob/f400007f77b756f6262070ef14a7476353359bfc/analyzers/src/SonarAnalyzer.Common/Helpers/TypeHelper.cs#L60) which gets called by `Is` does more checks on the type, which we don't need to do for most attributes.

I'd rather keep it as is, I don't think it's difficult to read.